### ISSUE TITLE
Invalid email addresses

### ISSUE THREAD

**Issue by davidearl** [PHPMailer/PHPMailer#428]
Hi, thank you for PHPMailer.

In a list of email addresses, I had one with a space in it that wasn't supposed to be there. This seems to have causes the AddAddress method to print an error message to stdout. This meant the JSON result of the request body for the request which sent the email was messed up and caused the javascript in the browser to fail (the $.getJSON jquery call gave an error which wouldn't make sense to an end user).

How can I capture such errors messages? It doesn't seem right that they go to stdout. (I could separately apply a regex to the emails before passing them, but presumably there are other examples).

**Comment by Synchro** [comment]
Neither `addAddress` or `validateAddress` output anything unless you have debug output enabled, but they do return false if adding or validating an address fails.

**Comment by davidearl** [comment]
Thanks - so how would debugging become enabled (or how would I turn it off)? I'm not aware of doing anything that would turn debugging on. It definitely seems to be coming from AddAddress.

**Comment by Synchro** [comment]
`addAddress` is a 1-line wrapper around `addAnAddress`, which is used by various other functions. It calls `edebug`, which only outputs anything if `$mail->SMTPDebug` > 0. You don't say what the error message is.

**Comment by davidearl** [comment]
OK, I found it - the error message was "Invalid address: <bit-of-address>" - it was indeed an echo in addAnAddress, but in an older version (I'd forgotten I'd taken it from another project rather than direct off GitHub). The current version doesn't have that, as you say, but does then fail completely quietly. However I noticed I can give TRUE to the constructor and have it throw an error, which fitted perfectly with the try..catch I already had. Sorry to have bothered you, I should have tried updating first.
